DifferentiationInterface/ext/DifferentiationInterfaceMooncakeExt/differentiate_with.jl

```diff
@@ -89,90 +110,93 @@ end
-        false,
-        :none,
-        nothing,
-        x -> +(x...),
-        randn(33),
-    ))
-    push!(
-        test_cases,
-        (
-            false,
-            :none,
-            nothing,
-            (
-                function (x)
-                    rx = Ref(x)
-                    return Base.pointerref(
-                        Base.bitcast(Ptr{Float64}, pointer_from_objref(rx)), 1, 1
-                    )
-                end
-            ),
-            5.0,
-        ),
-    )
-    push!(
-        test_cases,
-        (
-            false,
-            :none,
-            nothing,
-            x -> (pointerset(pointer(x), UInt8(3), 2, 1); x),
-            rand(UInt8, 5),
-        ),
-    )
-    push!(test_cases, (false, :none, nothing, Mooncake.__vec_to_tuple, [1.0]))
-    push!(test_cases, (false, :none, nothing, Mooncake.__vec_to_tuple, Any[1.0]))
-    push!(test_cases, (false, :none, nothing, Mooncake.__vec_to_tuple, Any[[1.0]]))
-    push!(test_cases, (false, :stability, nothing, Mooncake.IntrinsicsWrappers.ctlz_int, 5))
-    push!(
-        test_cases, (false, :stability, nothing, Mooncake.IntrinsicsWrappers.ctpop_int, 5)
-    )
-    push!(test_cases, (false, :stability, nothing, Mooncake.IntrinsicsWrappers.cttz_int, 5))
-    push!(
-        test_cases, (false, :stability, nothing, Mooncake.IntrinsicsWrappers.abs_float, 5.0)
-    )
-    push!(
-        test_cases,
-        (false, :stability, nothing, Mooncake.IntrinsicsWrappers.abs_float, 5.0f0),
-    )
-    push!(test_cases, (false, :stability, nothing, deepcopy, 5.0))
-    push!(test_cases, (false, :stability, nothing, deepcopy, randn(5)))
-    push!(test_cases, (false, :stability_and_allocs, nothing, sin, 1.1))
-    push!(test_cases, (false, :stability_and_allocs, nothing, sin, 1.0f1))
-    push!(test_cases, (false, :stability_and_allocs, nothing, cos, 1.1))
-    push!(test_cases, (false, :stability_and_allocs, nothing, cos, 1.0f1))
-    push!(test_cases, (false, :stability_and_allocs, nothing, exp, 1.1))
-    push!(test_cases, (false, :stability_and_allocs, nothing, exp, 1.0f1))
-
-    # additional_test_set = Mooncake.tangent_test_cases()
-    # function is_valid(f)
-    #     try
-    #         isa(f([1.0, 2.0]), Union{<:Number,<:AbstractArray})
-    #     catch
-    #         false
-    #     end
-    # end
-    # for test in additional_test_set
-    #     if is_valid(test[2])
-    #         push!(test_cases, test)
-    #     end
-    # end
+
+    map([(x) -> DI.DifferentiateWith(x, DI.AutoZygote())]) do F
+        map([Float64, Float32]) do P
+            push!(
+                test_cases,
+                Any[
+                    (false, :stability, nothing, F(Base.FastMath.sincos), P(3.0)),
+                    (false, :none, nothing, F(Mooncake.__vec_to_tuple), Any[P(1.0)]),
+                ]...,
+            )
+        end
+    end
+
+    map([(x) -> DI.DifferentiateWith(x, DI.AutoZygote())]) do F
```

Comment:
Better to avoid `Zygote`.

```suggestion
    map([(x) -> DI.DifferentiateWith(x, DI.ChainRules())]) do F
```